internal/rbd/controllerserver.go

```diff
@@ -266,15 +266,19 @@ func (rbdVol *rbdVolume) ToCSI(ctx context.Context) *csi.Volume {
 		}
 	}
 
-	return vol
+	return vol, nil
 }
 
 func buildCreateVolumeResponse(
 	ctx context.Context,
 	req *csi.CreateVolumeRequest,
 	rbdVol *rbdVolume,
 ) *csi.CreateVolumeResponse {
-	volume := rbdVol.ToCSI(ctx)
+	volume, err := rbdVol.ToCSI(ctx)
+	if err != nil {
+		log.ErrorLog(ctx, "BUG, can not happen: failed to convert volume %q to CSI type: %w", rbdVol, err)
```

Comment:
panic or return error here as well? continue will result in other problems

Reply:
Do you have a preference to panic or return an error? I guess returning an error is the cleanest.

Reply:
return error looks cleaner

Reply:
done